swap/types.go

```diff
@@ -49,3 +54,120 @@ type EmitChequeMsg struct {
 type ConfirmChequeMsg struct {
 	Cheque *Cheque
 }
+
+// Uint256 represents an unsigned integer of 256 bits
+type Uint256 struct {
+	value big.Int
+}
+
+var minUint256 = big.NewInt(0)
+var maxUint256 = new(big.Int).Sub(new(big.Int).Exp(big.NewInt(2), big.NewInt(256), nil), big.NewInt(1)) // 2^256 - 1
+
+// NewUint256 creates a Uint256 struct with a minimum initial underlying value
+func NewUint256() *Uint256 {
+	u := new(Uint256)
+	u.value = *minUint256
+	return u
+}
+
+// Uint64ToUint256 creates a Uint256 struct based on the given uint64 param
+// any uint64 is valid as a Uint256
+func Uint64ToUint256(base uint64) *Uint256 {
+	u := NewUint256()
+	u.value = *new(big.Int).SetUint64(base)
+	return u
+}
+
+// Value returns the underlying private value for a Uint256 struct
+func (u *Uint256) Value() big.Int {
+	return u.value
+}
+
+// Set assigns the underlying value of the given Uint256 param to u, and returns the modified receiver struct
+// returns an error when the result falls outside of the unsigned 256-bit integer range
+func (u *Uint256) Set(value big.Int) (*Uint256, error) {
+	if value.Cmp(maxUint256) == 1 {
+		return nil, fmt.Errorf("cannot set Uint256 to %v as it overflows max value of %v", value, maxUint256)
+	}
+	if value.Cmp(minUint256) == -1 {
+		return nil, fmt.Errorf("cannot set Uint256 to %v as it underflows min value of %v", value, minUint256)
+	}
+	u.value = value
+	return u, nil
+}
+
+// Copy sets the underlying value of u to a copy of the given Uint256 param, and returns the modified receiver struct
+func (u *Uint256) Copy(v *Uint256) *Uint256 {
```

Comment:
It is not clear what you want to achieve here; if it is a deep copy, then that is not what is happening, but basically the same as noted in `Set`

Reply:
correct